src/domain/core/src/messages/core_message_types.rs

```diff
@@ -23,8 +23,9 @@ const DATASET_LIFECYCLE_OUTBOX_VERSION: u32 = 1;
 pub enum DatasetLifecycleMessage {
     Created(DatasetLifecycleMessageCreated),
     DependenciesUpdated(DatasetLifecycleMessageDependenciesUpdated),
-    Deleted(DatasetLifecycleMessageDeleted),
     Renamed(DatasetLifecycleMessageRenamed),
+    About2Delete(DatasetLifecycleMessageAbout2Delete),
```

Comment:
Why do we need to separate "about to delete" and "deleted" messages?

I thought that one benefit of combining `immediate` and `durable` delivery mechanisms in one messaging system is that things that need to happen inside of the same transaction can subscribe to the same event as those that need to happen outside of the transaction.

Here it almost feels like `About2Delete` event is intended ONLY to be delivered via `immediate` mode, which seems strange as delivery is currently a property of a consumer.

Reply:
I've tried cascade deletions on foreign keys in new tables, and this worked better, so `AboutToDelete` is no longer needed.